platform-sdk/docs/proposals/state-lifecycle/proposal.md

```diff
@@ -33,66 +33,76 @@ These methods correspond to the state lifecycle as follows:
 This method should be moved to a separate class that will be responsible for managing the state lifecycle.
 
 `SwirldStateManager` needs refactoring. The current implementation has two sets of responsibilities:
-  - Keeping track of the references to the latest immutable state and the latest mutable state. This includes creating a copy of the state and updating the references. 
-  This functionality belongs to `swirlds-state-api`.   
-  
-  - Handling Platform-related events (belongs to `swirlds-platform-core`):
-    - Handling consensus rounds.
-    - Sealing consensus rounds.
-    - Keeping the `freeze time` parameter up-to-date.
-    - Providing state for signing.
-    - Initializing the state by extracting it from the signed state.
+- Keeping track of the references to the latest immutable state and the latest mutable state. This includes creating a copy of the state and updating the references.
+This functionality belongs to `swirlds-state-api`.
+
+- Handling Platform-related events (belongs to `swirlds-platform-core`):
+  - Handling consensus rounds.
+  - Sealing consensus rounds.
+  - Keeping the `freeze time` parameter up-to-date.
+  - Providing state for signing.
+  - Initializing the state by extracting it from the signed state.
 
 These two sets of responsibilities should be separated. The `swirlds-state-impl` should not have any details related to the platform. However, it should provide all necessary levers for the platform to interact with the state and its lifecycle.
 
-Therefore, `swirlds-state-api` needs a `StateManager` interface with an implementation `StateManagerImpl` in `swirlds-state-impl`  that will have the following responsibilities:
-- Create an empty state
-- Keeping track of the references to the latest immutable state and the latest mutable state.
-- Tracking states that existed before the latest immutable state. These should be available for lookup by their round number.
-- Restricts mutability to a single state. That is, only one state object should be mutable at any given time.
-- Evicting old states from memory that are no longer needed. `evictState` method *must* fail if the state is still mutable.
-- Loading snapshots from the disk.
+Therefore, `swirlds-state-api` needs a `StateLifecycleManager` interface with an implementation `StateLifecycleManagerImpl` in `swirlds-state-impl`  that will have the following responsibilities:
+- Take initial state as a starting point. It can be a genesis state or a state loaded from a snapshot.
+- Keep track of the references to the latest immutable state and the latest mutable state.
+- Restrict mutability to a single state. That is, only one state object should be mutable at any given time.
+
+The interface will look something like this:
+
+```java
+public interface StateLifecycleManager {
+    void setInitialState(State initialState);
```

Comment:
In the normal startup of a CN that has a state on disk, will this be how the state lifecycle manager gets initialized?

1. Load state from disk (I think this should be in the StateLifecycleManager, explained in a [comment](https://github.com/hashgraph/hedera-services/pull/16592/files/160388010a7ac16edfd958e431815dd40bb26080..5b058e400180e8d1a7fcc8c032de6f2072b56b7d#r1853936664) below
2. Use that state to invoke `setInitialState()` which then performs a fast copy, setting both the latest immutable state and the mutable state

Reply:
Yes, this makes sense. In this case there is no need to expose `setInitialState`, as it should be a responsibility of the `StateLifecycleManager` to deal with it

Reply:
I'll update the doc accordingly